nautobot_device_lifecycle_mgmt/jobs/model_migration.py

```diff
@@ -1064,154 +1069,94 @@ def migrate_content_type_references_to_new_model(self, old_ct, new_ct, hide_chan
             )
             object_permission.object_types.add(new_ct)
 
-        # These are migrated separately as they follow specific business logic
+    def migrate_relationships(self, content_types, debug):
+        """Migrate Relationships and RelationshipAssociations from the old models to the new models.
+
+        Relationships will be renamed to `{foo}_old` and a new relationship called `{foo}` will be created
+        with the same data and the `source_type`/`destination_type` will be updated with the new model's content type.
+        RelationshipAssociations will be duplicated and the content types updated as well.
+
+        """
+        # These relationships are migrated separately as they follow specific business logic
         excluded_relationships = ("device_soft", "inventory_item_soft")
         # Migrate Relationship content type
-        relationship_associations_dlm_source = []
-        for relationship in Relationship.objects.filter(source_type=old_ct).exclude(key__in=excluded_relationships):
-            self.logger.info(
-                "The Relationship __%s__ will be migrated to Core model __%s__",
-                html.escape(str(relationship.label)),
-                html.escape(str(new_ct)),
-            )
-            # We can't migrate relationships in place
-            try:
-                for relationship_association in RelationshipAssociation.objects.filter(
-                    source_type=old_ct, relationship=relationship
-                ).exclude(relationship__key__in=excluded_relationships):
-                    self.logger.info(
-                        "The Relationship Association __%s__ will be migrated to Core model __%s__",
-                        html.escape(str(relationship_association)),
-                        html.escape(str(new_ct)),
+        for old_relationship in Relationship.objects.filter(
+            Q(source_type__in=content_types.keys()) | Q(destination_type__in=content_types.keys())
+        ).exclude(key__in=excluded_relationships):
+            if (
+                old_relationship.key.endswith("_old")
+                and Relationship.objects.filter(key=old_relationship.key[:-4]).exists()
+            ) or (
+                old_relationship.label.endswith(" (old)")
+                and Relationship.objects.filter(label=old_relationship.label[:-6]).exists()
+            ):
```

Comment:
Not sure if we need to guard for the possibility of users manually updating the `label` to end in ` (old)` with key remaining unchanged.

Reply:
In the `Relationship` model, the `key` and `label` fields are each unique. So we need to check each field independently. If they were `unique_together(("key", "label"))` then we could check them together.

Reply:
I believe this is acceptable. With these being legacy fields and just here during the migration period, there should be a low risk to them changing.

Reply:
Makes sense, let's leave it as is.